doc/tutorials/dune-package-management/locking/dune-workspace

```diff
@@ -0,0 +1,13 @@
+(lang dune 3.21)
+(pkg enabled)
+
+(repository
+ (name overlay1)
+ (url "https://github.com/ocaml-dune/opam-overlays.git#2a9543286ff0e0656058fee5c0da7abc16b8717d"))
+
+(repository
+ (name upstream1)
+ (url "https://github.com/ocaml/opam-repository.git#2fd4164ca1e27b8c6027454c4844c1a1f6dca0bc"))
+
+(lock_dir
+ (repositories overlay1 upstream1))
```

Comment:
What is the semantics of the `lock_dir` stanza (with no specified name) when one doesn't have a `dune.lock` in the source? Or does it require you do have that in the source?